File: scribble_kiddies/keyPadCombis.py
keypad = {0:".",1:"abc",2:"def",3:"ghi",4:"jkl",5:"mno",6:"pqrs",7:"tu",8:"vwx",9:"yz"}
class Solution:
    def helper(self,keys,i,n,ans):
        global keypad
        if i == n:
            print(ans)
            return
        for j in keypad[int(keys[i])]:
            self.helper(keys,i+1,n,ans+j)

    def keyPadCombis(self,keys):
        self.helper(keys,0,len(keys),"")
    
A = Solution()
A.keyPadCombis("12")

File: scribble_kiddies/java_codes/keypad_combis.java
package java_codes;

public class keypad_combis {
    public static String[] keypad = { ".", "abc", "def", "ghi", "jkl", "mno", "pqrs", "uv", "wx", "yz" };

    public static void make(String S, int n, int i, String ans) {
        if (i == n) {
            System.out.println(ans);
            return;
        }
        int idx = S.charAt(i) - '0';
        for (int j = 0; j < keypad[idx].length(); j++) {
            make(S, n, i + 1, ans + keypad[idx].charAt(j));
        }

    }

    public static void main(String[] args) {
        make("12", 2, 0, "");
    }
}
